### PR TITLE
enable splay/force when in daemon mode but not using signal

### PR DESCRIPTION
**Pull request by olivierHa**
With this fix, now we can get a working "--force" or splay params, in the case when we are running puppet in daemon mode, but we don't want to use signal (to be able to give noop or environment variables)

### CODE REVIEW

**Comment by olivierHa** [comment]
CLA ok

**Comment by richardc** [comment]
Please update the commit message so that it (a) spells splay correctly, (b) includes the description of why you're changing what you're changing that's currently just in the pull request description.

**Comment by olivierHa** [comment]
Hello, 
It should be better.
Regards
Olivier

File: agent/puppet.rb [hunk 1]
```diff
@@ -202,8 +202,8 @@ def run(command, options)
         args[:ignoreschedules] = request[:ignoreschedules] if request[:ignoreschedules]
         args[:signal_daemon] = false if MCollective::Util.windows?
 
-        # we can only pass splay arguments if the daemon isn't running :(
-        unless @puppet_agent.status[:daemon_present]
+        # we can only pass splay arguments if the daemon isn't in signal mode :(
+        unless @puppet_agent.status[:daemon_present] && Util.str_to_bool(@config.pluginconf.fetch("puppet.signal_daemon","true"))
```
Comment (richardc) [inline agent/puppet.rb]: This is now really hard to follow, especially as it spills over 120 columns.  Can we hoist that condition into a value, with a better name, something like:

```
must_signal_daemon = @puppet_agent.status[:daemon_present] && Util.str_to_bool(@config.pluginconf.fetch("puppet.signal_daemon","true"))
unless must_signal_daemon
   ...
end
```

**Comment by olivierHa** [comment]
This should be more readable.

**Comment by richardc** [comment]
It reads better, now please squash into a single commit with an amended commit message.

**Comment by richardc** [comment]
You seem to have picked up a strat commit now.  I'll stage this to a clean branch and put it up.

**Comment by richardc** [comment]
https://github.com/puppetlabs/mcollective-puppet-agent/pull/59 is a tidied up version.  Thanks for the contribution.

**Comment by olivierHa** [comment]
Thanks !
